### DISCUSSION TITLE
kart hooks

### DISCUSSION THREAD

**Discussion by EwoudK** [koordinates/kart#980] [Q&A]
Hi
I was wondering if hooks are supported with kart? I have a github repository where I store my .gpkg data and I would like to amend a `kart diff` output to all my commit messages so I can keep track of what changes were made in which commit (which one can do with a pre-commit hook in a 'normal' code repository.

**Comment by olsen232** [comment]
Hi EwoudK,
In spite of appearances, Kart doesn't currently guarantee support of any hooks - the existence of all the samples in the `.kart/hooks/` directory is an oversight.

The pre-push hook does happen to work - you can edit that, it has an LFS hook in there already but that is only relevant for point-cloud and raster datasets. For vector datasets, that LFS hook won't be doing anything - rewrite as you see fit.
Other hooks likely do not work, although I haven't done a full audit of them all.

You can of course
a) manually append a kart diff to your commit messages, or even write a script that does so automatically
b) Run `kart show` on a commit at any time to see what changes it contains

**Reply by EwoudK** [reply]
Hi
Thanks for the reply. Currently I'm doing `kart diff | kart commit -m-` but I was wondering if there was something more official that would also work in combination with the qgis plugin. Does the pre-push hook also run when the push is made in qgis?

**Reply by olsen232** [reply]
I believe pre-push hooks also work from within the qgis plugin.

You can also run kart commands from outside qgis while qgis is running - you could make edits within QGIS and then run an external script to decide on a name for the current diffs, or even commit them.
Currently there is no official solution in this area - some hooks are working because Kart is using git directly, which supports hooks, and other hooks are not working because Kart is also using libgit2, which does not.
